src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -727,6 +748,37 @@ internal AnalyzerDriver(ImmutableArray<DiagnosticAnalyzer> analyzers, Func<Synta
             }
         }
 
+        private ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<CodeBlockAnalyzerAction>> CodeBlockActionsByAnalyzer
+        {
+            get
+            {
+                if (_lazyCodeBlockActionsByAnalyzer == null)
+                {
+                    ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<CodeBlockAnalyzerAction>> codeBlockActionsByAnalyzer;
+                    var codeBlockActions = this.analyzerActions.CodeBlockActions;
+                    if (codeBlockActions.Any())
```

Comment:
It's an immutable array, so you are correct, a simple property reference suffices.